tests/frontend/test_engine.py

```diff
@@ -166,209 +175,299 @@ def inspect():
 
-    def test__run_program(self, starship_engine, monkeypatch):
-        """
-        Tests StarshipEngine._run_program. Asserts that a program is converted to blackbird code,
-        compiled into a job content string and that the job is queued. Also asserts that a
-        completed job's result samples are returned.
-        """
-        mock_to_blackbird = MagicMock()
-        mock__get_blackbird = MagicMock()
-        program = MagicMock()
-        mock_job = MagicMock()
-        mock_job.is_complete = True
-        mock_job.is_failed = False
+    def test_get_job_error(self, connection, monkeypatch):
+        """Tests a failed job request."""
+        monkeypatch.setattr(requests, "get", mock_return(mock_response(404, {})))
+
+        with pytest.raises(RequestFailedError):
+            connection.get_job("123")
 
-        monkeypatch.setattr("strawberryfields.engine.to_blackbird", mock_to_blackbird)
-        monkeypatch.setattr(starship_engine, "_get_blackbird", mock__get_blackbird)
-        monkeypatch.setattr(starship_engine, "_queue_job", lambda job_content: mock_job)
+    def test_get_job_status(self, connection, monkeypatch):
+        """Tests a successful job status request."""
+        id_, status = "123", JobStatus.COMPLETED
 
-        some_params = {"param": MagicMock()}
-        result = starship_engine._run_program(program, args={}, compile_options={}, **some_params)
+        monkeypatch.setattr(
+            requests,
+            "get",
+            mock_return(mock_response(200, {"id": id_, "status": status.value})),
+        )
 
-        mock__get_blackbird.assert_called_once_with(program=program, param=some_params["param"],
-                args={}, compile_options={})
+        assert connection.get_job_status(id_) == status
 
-        assert result == mock_job.result.result.value
+    def test_get_job_status_error(self, connection, monkeypatch):
+        """Tests a failed job status request."""
+        monkeypatch.setattr(requests, "get", mock_return(mock_response(404, {})))
 
-    def test__run_program_fails(self, starship_engine, monkeypatch):
-        """
-        Tests that an Exception is raised when a job has failed.
-        """
-        mock_to_blackbird = MagicMock()
-        mock__get_blackbird = MagicMock()
-        program = MagicMock()
-        mock_job = MagicMock()
-        mock_job.is_complete = False
-        mock_job.is_failed = True
+        with pytest.raises(RequestFailedError):
+            connection.get_job_status("123")
 
-        monkeypatch.setattr("strawberryfields.engine.to_blackbird", mock_to_blackbird)
-        monkeypatch.setattr(starship_engine, "_get_blackbird", mock__get_blackbird)
-        monkeypatch.setattr(starship_engine, "_queue_job", lambda job_content: mock_job)
+    def test_get_job_result(self, connection, monkeypatch):
+        """Tests a successful job result request."""
+        result_samples = np.array([[1, 2], [3, 4]], dtype=np.int8)
```

Comment:
The `dtype` argument could be parametrized as well, just to see that it works fine for types like `np.float64` and the likes as well :)

Reply:
Good idea, will do so!

Reply:
This has been added!